pkg/cloudprovider/aws/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -73,12 +96,88 @@ func (e EKS) nodeLabelArg() string {
 	nodeLabelArg := ""
 	labelStrings := []string{}
 	var once sync.Once
-	for k, v := range e.Labels {
-		if v1alpha5.LabelDomainExceptions.Has(k) {
+	keys := lo.Keys(e.Labels)
+	sort.Strings(keys) // ensures this list is deterministic, for easy testing.
+	for _, key := range keys {
+		if v1alpha5.LabelDomainExceptions.Has(key) {
 			continue
 		}
 		once.Do(func() { nodeLabelArg = "--node-labels=" })
-		labelStrings = append(labelStrings, fmt.Sprintf("%s=%v", k, v))
+		labelStrings = append(labelStrings, fmt.Sprintf("%s=%v", key, e.Labels[key]))
 	}
 	return fmt.Sprintf("%s%s", nodeLabelArg, strings.Join(labelStrings, ","))
 }
+
+func (e EKS) mergeCustomUserData(userData bytes.Buffer) (bytes.Buffer, error) {
```

Comment:
Make this a pointer to avoid copying everywhere. That will help you follow the `nil,err` pattern elsewhere, too. 
```suggestion
func (e EKS) mergeCustomUserData(userData *bytes.Buffer) (*bytes.Buffer, error) {
```